### PR TITLE
feat: model client responses

### PR DESCRIPTION
**Pull request by booniepepper**
## Description

feat: model client responses

## References

Generated by https://github.com/openfga/sdk-generator/pull/218

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: src/main/java/dev/openfga/sdk/errors/FgaError.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        // FGA and OAuth2 servers are only expected to return HTTP 2xx responses.
+        if (isSuccessful(status)) {
+            return Optional.empty();
+        }
+
+        final String body = response.body();
+        final var headers = response.headers();
+        final FgaError error;
+
+        if (status == BAD_REQUEST || status == UNPROCESSABLE_ENTITY) {
```
Comment (mraible) [inline src/main/java/dev/openfga/sdk/errors/FgaError.java:30]: What's the baseline Java version for this SDK? If you can use the new `switch()` expression, you might be able to clean this up a bit. https://blogs.oracle.com/javamagazine/post/new-switch-expressions-in-java-12

Comment (booniepepper) [reply src/main/java/dev/openfga/sdk/errors/FgaError.java:30]: Java 11

(With apologies to anyone still using JDK 8)

File: src/main/java/dev/openfga/sdk/errors/HttpStatusCode.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    public static boolean isServerError(int statusCode) {
+        return isBetween(500, statusCode, 600);
+    }
+
+    /**
+     * Returns true for an HTTP status code that could reasonably be retried.
+     */
+    public static boolean isRetryable(int statusCode) {
+        return statusCode == 429 || (isServerError(statusCode) && statusCode != NOT_IMPLEMENTED);
```
Comment (rhamzeh) [inline src/main/java/dev/openfga/sdk/errors/HttpStatusCode.java:32]: Why not use the consts here and elsewhere in the file?

e.g.
```java
return statusCode == TOO_MANY_REQUESTS || (isServerError(statusCode) && statusCode != NOT_IMPLEMENTED);
```

**Review by rhamzeh — DISMISSED** [review]

File: src/main/java/dev/openfga/sdk/errors/HttpStatusCode.java (hunk 2 above)
Comment (booniepepper) [reply src/main/java/dev/openfga/sdk/errors/HttpStatusCode.java:32]: Oops, let me squeeze that in

Comment (booniepepper) [reply src/main/java/dev/openfga/sdk/errors/HttpStatusCode.java:32]: For the 2xx and 5xx I think the "between MIN/MAX" is more readable than enumerating the codes, but open to ideas

Comment (booniepepper) [reply src/main/java/dev/openfga/sdk/errors/HttpStatusCode.java:32]:
```suggestion
        return statusCode == TOO_MANY_REQUESTS || (isServerError(statusCode) && statusCode != NOT_IMPLEMENTED);
```

Comment (rhamzeh) [reply src/main/java/dev/openfga/sdk/errors/HttpStatusCode.java]: Works for now, we can always enhance it later on by adding more codes

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**